lib/mix/tasks/socket_proxy.ex

```diff
@@ -7,9 +7,17 @@ defmodule Mix.Tasks.SocketProxy do
   """
   def run(args) do
     {opts, ips, []} = OptionParser.parse(args, switches: [listen_port: :integer])
+    destinations = Enum.map(ips, &parse_ip_port/1)
     Application.put_env(:socket_proxy, :listen_port, opts[:listen_port])
-    Application.put_env(:socket_proxy, :destinations, ips)
+    Application.put_env(:socket_proxy, :destinations, destinations)
     {:ok, [:socket_proxy]} = Application.ensure_all_started(:socket_proxy)
     Process.sleep(:infinity)
   end
+
+  defp parse_ip_port(ip_port) do
+    [raw_host, raw_port] = String.split(ip_port, ":")
```

Comment:
Oh, good question. I didn't think to look at `URI` for helping here. It seems like `URI.parse("127.0.0.1:8000")` chooses to parse that into the `path` field, for some reason, although `URI.parse("//127.0.0.1:8000")` does parse it into a host of `"127.0.0.1"` and port of `8000`, so that's nice, if a bit weird. I think it's probably not worth it, since `:gen_tcp` expects the IP address in the form `{127, 0, 0, 1}`, so we'd still need to do much of this `parse_ip_port` work anyway (the `String.to_charlist` and `:inet.parse_address`).

In answer to your first question, no, ports aren't part of an IP address. Considering the [OSI Model](https://github.com/mbta/logger_splunk_backend), IP addresses are a part of the Network (e.g. IP) layer, responsible for getting data routed across the internet, while the ports are part of the Transport (e.g. TCP) layer.